homeassistant/components/mutesync/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for mütesync integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultDict
+from homeassistant.exceptions import HomeAssistantError
+
+import mutesync
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# TODO adjust the data schema to the data that you need
+STEP_USER_DATA_SCHEMA = vol.Schema({"host": str})
+
+
+class PlaceholderHub:
+    """Placeholder class to make tests pass.
+
+    TODO Remove this placeholder class and replace with things from your PyPI package.
+    """
+
+    def __init__(self, host: str) -> None:
+        """Initialize."""
+        self.host = host
+
+    async def authenticate(self, username: str, password: str) -> bool:
+        """Test if we can authenticate with the host."""
+        return True
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    session = hass.helpers.aiohttp_client.async_get_clientsession()
+    try:
+        token = await mutesync.authenticate(session, data["host"])
+    except (aiohttp.ClientError, asyncio.TimeoutError):
+        raise CannotConnect
```

Comment:
It would be great if you could differentiate between a "authentication is not open" vs "cannot connect" so the user can be instructed accordingly in the UI.

Reply:
That sounds like a good idea.

I'm thinking mutesync can return a `403` if authentication is not open. Is that fine by you?

How would I catch a `403` and raise an `InvalidAuth`?

I can catch and process `InvalidAuth` in `async_setup_user`. Is that fine?